node/src/main/scala/coop/rchain/node/proposescript/ProposeConfig.scala

```diff
@@ -0,0 +1,132 @@
+
+  def checkLeaderBlockInFollower[F[_]: Concurrent: Monixable: Log](blockHash: String): F[Boolean] =
+    Stream
+      .emits(followers)
+      .parEvalMapUnordered[F, Boolean](followers.length)(
+        v =>
+          v.isContainBlock(blockHash).flatTap { isContain =>
+            v.logValidatorGotBlock(blockHash)
+              .whenA(isContain) >> v.logValidatorMissedBlock(blockHash).whenA(!isContain)
+          }
+      )
+      .forall(identity)
+      .compile
+      .last
+      .map(_.getOrElse(false))
+
+  def waitFollowersGotLeaderBlock[F[_]: Concurrent: Monixable: Log: Sleep](
+      blockHash: String
+  ): F[Boolean] = {
+    val retryPolicy = RetryPolicies.constantDelay(checkInterval)
+    def onFailure(failedValue: Boolean, details: RetryDetails): F[Unit] =
+      Log[F].info(
+        s"Retry check in ${checkInterval} because followers didn't get the block ${blockHash} from leader." +
+          s"Retry ${details.retriesSoFar} times so far, cumulate cumulativeDelay:${details.cumulativeDelay}, ${failedValue}."
+      )
+    retryingOnFailures[Boolean](
+      policy = retryPolicy,
+      wasSuccessful = identity,
+      onFailure = onFailure
+    ) {
+      checkLeaderBlockInFollower(blockHash)
+    }
+  }
+
+  def proposeInFollowers[F[_]: Concurrent: Monixable: Log]: F[List[String]] =
+    Stream
+      .emits(followers)
+      .parEvalMapUnordered[F, String](followers.length)(
+        v => v.deployAndProposeAsync.flatTap(blockHash => v.logValidatorProposedBlock(blockHash))
+      )
+      .compile
+      .toList
+
+  def checkFollowerBlocksInLeader[F[_]: Concurrent: Monixable: Log](
+      blockHashes: List[String]
+  ): F[Boolean] =
+    Stream
+      .emits(blockHashes)
+      .parEvalMapUnordered[F, Boolean](blockHashes.length)(
+        b =>
+          leader
+            .isContainBlock(b)
+            .flatTap { isContain =>
+              leader
+                .logValidatorGotBlock(b)
+                .whenA(isContain) >> leader.logValidatorMissedBlock(b).whenA(!isContain)
+            }
+      )
+      .forall(identity)
+      .compile
+      .last
+      .map(_.getOrElse(false))
+
+  def waitLeaderGotFollowerBlocks[F[_]: Concurrent: Monixable: Log: Sleep](
+      blocks: List[String]
+  ): F[Boolean] = {
+    val retryPolicy = RetryPolicies.constantDelay(checkInterval)
+    def onFailure(failedValue: Boolean, details: RetryDetails): F[Unit] =
+      Log[F].info(
+        s"Retry check in ${checkInterval} because leader didn't get the block from followers." +
+          s"Retry ${details.retriesSoFar} times so far, cumulate cumulativeDelay:${details.cumulativeDelay}, ${failedValue}."
+      )
+    retryingOnFailures[Boolean](
+      policy = retryPolicy,
+      wasSuccessful = identity,
+      onFailure = onFailure
+    ) {
+      checkFollowerBlocksInLeader(blocks)
+    }
+  }
+
+  def proposeRound[F[_]: Concurrent: Monixable: Log: Sleep]: F[Unit] =
```

Comment:
I'd put this function on top because it's definition of the whole process. It's like specification which I want to see first and from there explore inner components.